src/suites/cts/validation.spec.ts

```diff
@@ -0,0 +1,673 @@
+export const description = `
+Validation tests.
+`;
+
+import { TestGroup } from '../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+export const g = new TestGroup(ValidationTest);
+
+g.test('validation/binding count mismatch', async t => {
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  });
+
+  t.device.createBindGroup({
+    bindings: [],
+    layout: bindGroupLayout,
+  });
+
+  await t.expectUncapturedError();
+});
+
+g.test('validation/binding must be present in layout', async t => {
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
+        visibility: GPUShaderStage.COMPUTE,
+        type: 'storage-buffer',
+      },
+    ],
+  });
+
+  t.device.createBindGroup({
+    bindings: [
+      {
+        binding: 1,
+        resource: {
+          buffer: t.storageBuffer,
+        },
+      },
+    ],
+    layout: bindGroupLayout,
+  });
+
+  await t.expectUncapturedError();
+});
+
+g.test('validation/binding index too high in bind group', async t => {
+  const bindingIndexTooHigh = 16;
+
+  const bindGroupLayout = t.device.createBindGroupLayout({
+    bindings: [
+      {
+        binding: 0,
```

Comment:
Since the layout has binding 0 and the bind group has binding 16, this would fail even if 16 was valid. Change this to bindingIndexTooHigh?

Reply:
@kainino0x Good catch!
Should we even have a test then? Since we already have one for createBindGroupLayout that tests bindingIndexTooHigh.